server/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java

```diff
@@ -475,4 +477,23 @@ public Set<String> getIndices() {
     public boolean isSimulated() {
         return false; // Always false, but may be overridden by a subclass
     }
+
+    /*
+     * Returns any component template substitutions that are to be used as part of this bulk request. We would likely only have
+     * substitutions in the event of a simulated request.
+     */
+    public Map<String, ComponentTemplate> getComponentTemplateSubstitutions() throws IOException {
+        return Map.of();
+    }
+
+    /*
+     * This copies this bulk request, but without all of its inner requests
+     */
+    public BulkRequest shallowClone() {
+        BulkRequest bulkRequest = new BulkRequest();
+        bulkRequest.setRefreshPolicy(getRefreshPolicy());
+        bulkRequest.waitForActiveShards(waitForActiveShards());
+        bulkRequest.timeout(timeout());
+        return bulkRequest;
```

Comment:
`BulkRequest` has many other things than only the refresh policy, active shards, and timeout. Are we going to copy the rest of them also? Should the javadoc reflect what's being copied?

Reply:
I went his way because this is what BulkRequestModifier did (this code was moved out of there). I wavered on whether I ought to add the other fields or not, and wound up just repeating and testing the previous behavior. Any thoughts?

Reply:
I think just documenting it in the javadoc is fine

Reply:
I wound up adding the rest of the fields just in case (I don't think they'd intentionally been left out -- I think they were just added after the BulkRequestModifier method had been written).